### PR TITLE
Nc news 11

### PR DESCRIPTION
**Pull request by JackCollier**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -123,14 +123,54 @@ describe("GET /api/articles", () => {
...
     return request(app)
       .get("/api/articles")
       .then(({ body }) => {
         const { articles } = body;
         expect(articles).toBeSortedBy("created_at", { descending: true });
       });
   });
+  test("should be filtered by topic mitch", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
```
Comment (johnhutcheon) [inline __tests__/api.test.js:136]: Ensure you're expecting a 200 status for each of these tests with a valid query, you've missed a few in here 👍

File: __tests__/api.test.js [hunk 2]
```diff
@@ -123,14 +123,54 @@ describe("GET /api/articles", () => {
...
+  test("should be sorted by title", () => {
+    return request(app)
+      .get("/api/articles?sort_by=title")
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("title", { descending: true });
+      });
+  });
+  test("should return 404 status when passed a non existant sort_by", () => {
+    return request(app).get("/api/articles?sort_by=jeff").expect(404);
```
Comment (johnhutcheon) [inline __tests__/api.test.js:161]: I can see why you've gone for a 404 here, but we'd actually expect a 400 in this instance. We can expect a 404 with the topic, as that topic may one day exist, where as with the sort_by, it's either a correct or incorrect query that we pass into the filter 👍

File: __tests__/api.test.js [hunk 3]
```diff
@@ -123,14 +123,54 @@ describe("GET /api/articles", () => {
...
+  test("should be ordered in ascedning", () => {
+    return request(app)
+      .get("/api/articles?order=ASC")
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("created_at", { descending: false });
+      });
+  });
+  test("should return 400 status when passed a bad order", () => {
+    return request(app).get("/api/articles?order=up").expect(404);
```
Comment (johnhutcheon) [inline __tests__/api.test.js:172]: You've described this test as expecting a 400, but the test expects a 404. Could you please amend so it expects a 400?

File: __tests__/api.test.js [hunk 4]
```diff
@@ -123,14 +123,54 @@ describe("GET /api/articles", () => {
...
+  });
 });
```
Comment (johnhutcheon) [inline __tests__/api.test.js:175]: We're just missing one last test. If we have a valid topic that has no articles, we should get an empty array. Could you please add this?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Jack, nice work! Could you just see my comments please and make the amendments to the tests as requested? Many thanks! 👍

**Review by roisinoreilly — APPROVED** [review]
nicely refactored, if I was being picky I would suggest another test which chains all 3 of your possible queries together to make sure the full query string is working, other than that feel free to merge! :)

**Pull request merged**